### PR TITLE
[Internal] Contract validation: Adds method info to the contract

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

This adds method info like IsVirtual to the public contract files. This will prevent accidental breaking changes such as removing a virtual from a method.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by ealsur — APPROVED** [review]
Nice job!

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcement.cs [hunk 1]
```diff
@@ -78,6 +78,42 @@ private static string GenerateNameWithClassAttributes(Type type)
...
+                $"{nameof(methodInfo.IsConstructor)}:{(methodInfo.IsConstructor ? bool.TrueString : bool.FalseString)};" +
+                $"{nameof(methodInfo.IsFinal)}:{(methodInfo.IsFinal ? bool.TrueString : bool.FalseString)};";
+        }
+
+        private static string GenerateNameWithPropertyAttributes(PropertyInfo propertyInfo)
+        {
+            string name = $"{propertyInfo};{nameof(propertyInfo.CanRead)}:{(propertyInfo.CanRead ? bool.TrueString : bool.FalseString)};" +
+                $"{nameof(propertyInfo.CanWrite)}:{(propertyInfo.CanWrite ? bool.TrueString : bool.FalseString)};";
+
+            MethodInfo getMethodInfo = propertyInfo.GetGetMethod();
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcement.cs:96]: nit: GetAccessors I think covers both

**Review by ealsur — APPROVED** [review]
Left only one nit, LGTM

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcement.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Contracts/ContractEnforcement.cs:96]: This is a cleaner model. GetAccessors returns an array which then has to be looped over. This makes it clear one is the get and other is the set accessor.

**Pull request merged**